### PR TITLE
change silent error to throw exception

### PR DESCRIPTION
**Pull request by jdorando**

### CODE REVIEW

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRCollector.java [hunk 1]
```diff
@@ -111,10 +112,8 @@ public TopDocs topDocs(int start, int howMany) {
         topRerankDocs = reRankRescorer.rescore(searcher,
             mainDocs, howMany);
       } catch (final IOException e) {
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRCollector.java]: Just remove the try catch, no need to rethrow the same exception

Comment (diegoceccarelli) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRCollector.java]: +1 This is not an exception produced by the LTR plugin, it is a Solr problem

**Comment by jdorando** [comment]
Updated for code review comments

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRCollector.java [hunk 2]
```diff
@@ -106,16 +106,8 @@ public TopDocs topDocs(int start, int howMany) {
...
-      try {
-        topRerankDocs = reRankRescorer.rescore(searcher,
-            mainDocs, howMany);
-      } catch (final IOException e) {
-        log.error("LTRRescorer reranking failed. ", e);
-        // If someone deployed a messed up model, we don't want to crash and burn.
-        // Return the original list at least
-        topRerankDocs = mainDocs;
-      }
+      TopDocs topRerankDocs = reRankRescorer.rescore(searcher,
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRCollector.java]: Add a unittest that verifies the new behavior.

**Comment by jdorando** [comment]
unit test added

**Pull request merged**